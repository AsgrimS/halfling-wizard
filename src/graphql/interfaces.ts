import { Class } from "./schema";

export interface HeroClassesData {
  classes: Class[];
}

export interface HeroClassData {
  class: Class;
}
