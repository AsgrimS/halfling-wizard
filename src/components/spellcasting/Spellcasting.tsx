import React from "react";

import { ClassSpellcastingData } from "../../graphql/interfaces";

const Spellcasting: React.FC<ClassSpellcastingData> = ({ spellcasting }) => {
  return (
    <div data-testid="spellcasting">
      <span>Spellcasting Ability:</span>
      <ul>
        <li>{spellcasting.spellcasting_ability?.name}</li>
      </ul>
    </div>
  );
};

export default Spellcasting;
