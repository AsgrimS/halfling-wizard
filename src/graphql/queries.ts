import { gql } from "@apollo/client";

export const HERO_CLASS = gql`
  query GetHeroesClasses($classIndex: String!) {
    class(filter: { name: $classIndex }) {
      hit_die
      index
      name
      proficiencies {
        index
        name
      }
      saving_throws {
        index
        name
      }
      proficiency_choices {
        choose
        from {
          name
          index
        }
      }
      spellcasting {
        level
        spellcasting_ability {
          index
          name
        }
      }
      starting_equipment {
        equipment {
          name
          index
        }
        quantity
      }
      starting_equipment_options {
        choose
        from {
          equipment {
            name
            index
          }
          quantity
        }
      }
      subclasses {
        index
        name
      }
    }
  }
`;

export const HEROES_CLASSES = gql`
  query GetHeroesClasses {
    classes {
      name
      index
    }
  }
`;
