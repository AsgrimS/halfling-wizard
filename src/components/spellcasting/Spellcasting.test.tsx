import React from "react";
import { render, screen, cleanup } from "@testing-library/react";

import Spellcasting from "./Spellcasting";
import { ClassSpellcasting } from "../../graphql/schema";

describe("Spellcasting component", () => {
  afterEach(() => {
    cleanup();
  });

  it("should render a name of spellcasting ability", () => {
    const mockSpellcastingData: ClassSpellcasting = {
      spellcasting_ability: {
        name: "INT",
      },
    };

    render(<Spellcasting spellcasting={mockSpellcastingData} />);
    const spellcastingElement = screen.getByTestId("spellcasting");
    const liElement = spellcastingElement.getElementsByTagName("li")[0];

    expect(spellcastingElement).toBeInTheDocument();
    expect(liElement).toContainHTML(
      `<li>${mockSpellcastingData.spellcasting_ability!.name}</li>`
    );

    expect(spellcastingElement).toMatchSnapshot();
  });
});
