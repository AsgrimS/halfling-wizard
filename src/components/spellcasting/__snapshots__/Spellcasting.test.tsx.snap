// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spellcasting component should render a name of spellcasting ability 1`] = `
<div
  data-testid="spellcasting"
>
  <span>
    Spellcasting Ability:
  </span>
  <ul>
    <li>
      INT
    </li>
  </ul>
</div>
`;
